app/models/theft_alert.rb

```diff
@@ -0,0 +1,76 @@
+class TheftAlert < ActiveRecord::Base
+  enum status: { pending: 0, active: 1, inactive: 2 }.freeze
+
+  validates :stolen_record,
+            :theft_alert_plan,
+            :status,
+            :creator,
+            presence: true
+
+  belongs_to :stolen_record
+  belongs_to :theft_alert_plan
+  belongs_to :payment
+  belongs_to :creator,
+             class_name: "User",
+             foreign_key: :user_id
+
+  scope :due_for_expiration, -> { active.where('"theft_alerts"."end_at" <= ?', Time.current) }
```

Comment:
nit: I think you can just do `active.where("end_at <= ?", Time.current)`

Reply:
You can, but I usually default to explicitness in scopes to aid composability — this sort of exception can be a hassle to debug when it's deep inside a query:

```rb
Bike.joins(:manufacturer).where('created_at < ?', Time.current)

# #<ActiveRecord::StatementInvalid: PG::AmbiguousColumn: ERROR:  column reference "created_at" is ambiguous

Bike.joins(:manufacturer).where('"bikes"."created_at" < ?', Time.current)

# SELECT COUNT(*) FROM "bikes" INNER JOIN "manufacturers" ON "manufacturers"."id" = "bikes"."manufacturer_id" WHERE "bikes"."example" = $1 AND "bikes"."hidden" = $2 AND ("bikes"."created_at" < '2019-06-19 23:55:12.317688')  [["example", "f"], ["hidden", "f"]]
#  Bike Load (736.8ms)  SELECT "bikes".* FROM "bikes" INNER JOIN "manufacturers" ON "manufacturers"."id" = "bikes"."
```